src/robot/conf/languages.py

```diff
@@ -63,8 +63,9 @@ def __init__(self, languages: 'Iterable[LanguageLike]|LanguageLike|None' = (),
     @property
     def bdd_prefix_regexp(self):
         if not self._bdd_prefix_regexp:
-            prefixes = '|'.join(self.bdd_prefixes).replace(' ', r'\s').lower()
-            self._bdd_prefix_regexp = re.compile(rf'({prefixes})\s', re.IGNORECASE)
+            prefixes = sorted(self.bdd_prefixes, key=len, reverse=True)
+            pattern = '|'.join(prefix.replace(' ', r'\s') for prefix in prefixes).lower()
+            self._bdd_prefix_regexp = re.compile(rf'({pattern})\s', re.IGNORECASE)
```

Comment:
It would be good to add a unit test for this fix in `utest/api/test_languages.py `.